google-startup-scripts/usr/share/google/onboot

```diff
@@ -112,6 +112,41 @@ function print_ssh_key_fingerprints() {
   return 0
 }
 
+function check_for_connection() {
+  local count=0
+  local return_code=0
+  local url="http://metadata/computeMetadata/v1/instance/?recursive=True"
```

Comment:
Same comment about choice of hostname as on get_metadata_value line 18 - either use ```169.264.169.254``` or ```metadata.google.internal``` but not ```metadata```.

Also, it would be great not to have to repeat this URL in multiple places. Do we have a reason to reimplement get_metadata_value in onboot? If get_metadata_value is not already general enough to avoid this, can we easily make it so?